geonode_mapstore_client/client/js/components/DetailsPanel/DetailsTimeSeries.jsx

```diff
@@ -0,0 +1,154 @@
+/*
+ * Copyright 2025, GeoSolutions Sas.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+import React, { useEffect, useState } from 'react';
+import PropTypes from "prop-types";
+import isNil from 'lodash/isNil';
+import isEmpty from 'lodash/isEmpty';
+import { Checkbox, FormGroup, ControlLabel, FormControl, HelpBlock } from 'react-bootstrap';
+import Select from 'react-select';
+import Message from '@mapstore/framework/components/I18N/Message';
+import HTML from '@mapstore/framework/components/I18N/HTML';
+import { TIME_SERIES_PROPERTIES, TIME_ATTRIBUTE_TYPES, TIME_PRECISION_STEPS } from '@js/utils/ResourceUtils';
+import { getMessageById } from '@mapstore/framework/utils/LocaleUtils';
+import InfoPopover from '@mapstore/framework/components/widgets/widget/InfoPopover';
+
+const TimeSeriesSettings = ({ resource, onChange }, context) => {
+    const timeAttributes = (resource?.attribute_set ?? [])
+        .filter((attribute) => TIME_ATTRIBUTE_TYPES.includes(attribute.attribute_type))
+        .map((attribute)=> ({value: attribute.pk, label: attribute.attribute}));
+
+    if (isEmpty(timeAttributes)) return null;
+
+    const [timeseries, setTimeSeries] = useState();
+    const [error, setError] = useState();
+    useEffect(() => {
+        resource.timeseries && setTimeSeries(resource.timeseries);
+    }, [resource?.timeseries]);
```

Comment:
probably we don't need a useEffect here, could you take a look if this kind of [approach](https://react.dev/learn/you-might-not-need-an-effect#adjusting-some-state-when-a-prop-changes) could work?